### PR TITLE
[EXP-78-278] Gearbox

### PR DESCRIPTION
**Pull request by AmantayAbdurakhmanov**

### CODE REVIEW

**Comment by BobTheBuidler** [comment]
I've tested this, works fine. 
@banteg could you also validate the math here since the stakes are high?

**Comment by BobTheBuidler** [comment]
@AmantayAbdurakhmanov I committed one more change to your fork after reviewing with @mariuspod , everything looks good to go now. Could you rebase your branch on master once more and then I can merge? Thanks!

File: yearn/partners/snapshot.py [hunk 1]
```diff
@@ -182,6 +182,36 @@ def unwrap(self) -> List[Wrapper]:
...
+                    [
+                        vault.vault,
+                        'balanceOf',
+                        ca,
+                        block
+                    ]
+                    for ca in CAs
+                ]
+            )   
+            tvl = sum(balance for balance in balances_at_block) / Decimal(vault.scale)
```
Comment (mariuspod) [inline yearn/partners/snapshot.py]: change to `sum(balance / Decimal(vault.scale) for balance in balances_at_block)`

**Review by mariuspod — CHANGES_REQUESTED** [review]

**Review by mariuspod — APPROVED** [review]
LGTM!

**Pull request merged**